slsim/Deflectors/cluster_catalog_lens.py

```diff
@@ -0,0 +1,281 @@
+import numpy as np
+import numpy.random as random
+from slsim.selection import object_cut
+from slsim.Deflectors.richness2mass import mass_richness_relation
+from slsim.Deflectors.halo_population import gene_e_ang_halo, concent_m_w_scatter
+from colossus.cosmology import cosmology as colossus_cosmo
+from slsim.Deflectors.velocity_dispersion import vel_disp_abundance_matching
+from slsim.Deflectors.elliptical_lens_galaxies import (
+    elliptical_projected_eccentricity,
+    vel_disp_from_m_star,
+)
+from slsim.Deflectors.velocity_dispersion import vel_disp_nfw
+from slsim.Deflectors.deflectors_base import DeflectorsBase
+from lenstronomy.Util.param_util import phi_q2_ellipticity
+from astropy import units as u
+from astropy.table import hstack
+from scipy.spatial.distance import cdist
+
+
+class ClusterCatalogLens(DeflectorsBase):
+    """Class describing cluster lens model with a NFW profile for the dark matter halo
+    and EPL profile for the subhalos (cluster members). It makes use of a group/cluster
+    catalog and a group/cluster member catalog (e.g. redMaPPer).
+
+    This class is called by setting deflector_type == "cluster-catalog" in LensPop.
+    """
+
+    def __init__(
+        self,
+        cluster_list,
+        members_list,
+        galaxy_list,
+        kwargs_cut,
+        kwargs_mass2light,
+        cosmo,
+        sky_area,
+        richness_fn="Abdullah2022",
+    ):
+        """
+
+        :param cluster_list: list of dictionary with lens parameters of
+            elliptical dark matter halos from a group/cluster catalog.
+            Mandatory keys: 'cluster_id' 'z', 'richness'
+        :param members_list: list of dictionary with lens parameters of
+            elliptical galaxies from a group/cluster member catalog.
+            Mandatory keys: 'cluster_id', 'ra', 'dec', 'mag_{band}'
+        :param kwargs_cut: cuts in parameters: band, band_mag, z_min, z_max
+        :type kwargs_cut: dict
+        :param kwargs_mass2light: mass-to-light relation
+        :param cosmo: astropy.cosmology instance
+        :type sky_area: `~astropy.units.Quantity`
+        :param sky_area: Sky area over which galaxies are sampled. Must be in units of
```

Comment:
For sky_area, we use astropy Quantity(). Is this astropy Quantity?

Reply:
Yes, it is